### PR TITLE
Upgrade to 16.0

### PR DESCRIPTION
**Pull request by wm-cortex**
Due to the changes in Odoo Oline payment API, I have upgraded it to Odoo 16.0 with the new payment API.

### CODE REVIEW

**Comment by wm-cortex** [comment]
@KangLeian Could you please test the upgraded app from your side?

File: models/payment_transaction.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        super()._process_notification_data(notification_data)
+        if self.provider_code != 'duitku':
+            return
+
+        self.duitku_reference = self.provider_reference= notification_data.get('reference')
+        self.duitku_order_id= notification_data.get('merchantOrderId')
+
+        payment_status = notification_data.get('resultCode')
+
+        if payment_status == '00':
```
Comment (KangLeian) [inline models/payment_transaction.py:244]: Hi @wm-cortex  , Please add signature validation and check transaction validation before processing here.

**Review by KangLeian — CHANGES_REQUESTED** [review]
Adding a validation before changing the status payment.

**Review by KangLeian — APPROVED** [review]
Awesome, Thank you for the updates!

**Comment by wm-cortex** [comment]
@KangLeian Thanks for merging.

**Pull request merged**